#!/usr/bin/env python

PACKAGE = "quad_flight"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("gain_position_p", double_t, 0, "Proportional position gain", 2.0, 0.0, 10.0)
gen.add("gain_orientation_p", double_t, 0, "Proportional orientation gain", 2.0, 0.0, 10.0)
gen.add("gain_position_i", double_t, 0, "Integral position gain", 0.002, 0.0, 10.0)
gen.add("gain_orientation_i", double_t, 0, "Integral orientation gain", 0.002, 0.0, 10.0)
gen.add("position_error_integral_lim", double_t, 0, "Integral limit for position", 2.0, 0.0, 10.0)
gen.add("orientation_error_integral_lim", double_t, 0, "Integral limit for position", 2.0, 0.0, 10.0)

exit(gen.generate(PACKAGE, "quad_flight", "PositionController"))

